@jacobperron No, I'm referring to this:

```
Starting >>> depth_image_proc
--- stderr: image_proc
/usr/bin/ld: /opt/ros/rolling/lib/libimage_transport.so: undefined reference to `rclcpp::SerializedMessage::SerializedMessage(unsigned long, rcutils_allocator_s const&)'
/usr/bin/ld: /opt/ros/rolling/lib/libimage_transport.so: undefined reference to `rclcpp::UnsupportedEventTypeException::UnsupportedEventTypeException(int, rcutils_error_state_s const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: /opt/ros/rolling/lib/libimage_transport.so: undefined reference to `rclcpp::exceptions::throw_from_rcl_error(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rcutils_error_state_s const*, void (*)())'
collect2: error: ld returned 1 exit status
```

--- a/image_view/include/image_view/image_saver_node.hpp
+++ b/image_view/include/image_view/image_saver_node.hpp
@@ -70,12 +70,13 @@ public:
 
 private:
   boost::format g_format;
-  bool save_all_image, save_image_service;
-  std::string encoding;
-  bool request_start_end;
-  bool is_first_image_;
-  bool has_camera_info_;
-  size_t count_;
+  bool save_all_image_{false};
+  bool save_image_service_{false};
+  std::string encoding_;
+  bool request_start_end_{false};
+  bool is_first_image_{true};
+  bool has_camera_info_{false};
+  size_t count_{0u};
   rclcpp::Time start_time_;
   rclcpp::Time end_time_;
   image_transport::CameraSubscriber cam_sub_;

--- a/image_view/src/image_saver_node.cpp
+++ b/image_view/src/image_saver_node.cpp
@@ -71,8 +71,7 @@ using std::placeholders::_2;
 using std::placeholders::_3;
 
 ImageSaverNode::ImageSaverNode(const rclcpp::NodeOptions & options)
-: rclcpp::Node("image_saver_node", options),
-  is_first_image_(true), has_camera_info_(false), count_(0)
+: rclcpp::Node("image_saver_node", options)
 {
   auto topic = rclcpp::expand_topic_or_service_name(
     "image", this->get_name(), this->get_namespace());
@@ -91,8 +90,9 @@ ImageSaverNode::ImageSaverNode(const rclcpp::NodeOptions & options)
 
   std::string format_string;
   format_string = this->declare_parameter("filename_format", std::string("left%04i.%s"));
-  encoding = this->declare_parameter("encoding", std::string("bgr8"));
-  save_all_image = this->declare_parameter("save_all_image", true);
+  encoding_ = this->declare_parameter("encoding", std::string("bgr8"));
+  save_all_image_ = this->declare_parameter("save_all_image", true);
+  request_start_end_ = this->declare_parameter("request_start_end", false);
   g_format.parse(format_string);
 
   save_srv_ = this->create_service<std_srvs::srv::Empty>(
@@ -108,11 +108,11 @@ bool ImageSaverNode::saveImage(
 {
   cv::Mat image;
   try {
-    image = cv_bridge::toCvShare(image_msg, encoding)->image;
+    image = cv_bridge::toCvShare(image_msg, encoding_)->image;
   } catch (const cv_bridge::Exception &) {
     RCLCPP_ERROR(
       this->get_logger(), "Unable to convert %s image to %s",
-      image_msg->encoding.c_str(), encoding.c_str());
+      image_msg->encoding.c_str(), encoding_.c_str());
     return false;
   }
 
@@ -135,11 +135,11 @@ bool ImageSaverNode::saveImage(
       g_format.clear();
     }
 
-    if (save_all_image || save_image_service) {
+    if (save_all_image_ || save_image_service_) {
       cv::imwrite(filename, image);
       RCLCPP_INFO(this->get_logger(), "Saved image %s", filename.c_str());
 
-      save_image_service = false;
+      save_image_service_ = false;
     } else {
       return false;
     }
@@ -159,7 +159,7 @@ bool ImageSaverNode::service(
   (void)request_header;
   (void)request;
   (void)response;
-  save_image_service = true;
+  save_image_service_ = true;
   return true;
 }
 
@@ -210,7 +210,7 @@ void ImageSaverNode::callbackWithoutCameraInfo(
   //  1. request by service.
   //  2. request by topic about start and end.
   //  3. flag 'save_all_image'.
-  if (!save_image_service && request_start_end) {
+  if (!save_image_service_ && request_start_end_) {
     if (start_time_ == rclcpp::Time(0)) {
       return;
     } else if (start_time_ > image_msg->header.stamp) {
@@ -235,7 +235,7 @@ void ImageSaverNode::callbackWithCameraInfo(
 {
   has_camera_info_ = true;
 
-  if (!save_image_service && request_start_end) {
+  if (!save_image_service_ && request_start_end_) {
     if (start_time_ == rclcpp::Time(0)) {
       return;
     } else if (start_time_ > image_msg->header.stamp) {

